### PR TITLE
new api module; aggregate and getDimensionMembers have been implemented

### PR DESCRIPTION
**Pull request by borysyuk**
this is not all functionality of this module but now there all what I and Levko need to working on chart components.

### CODE REVIEW

File: src2/api/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class Api {
+  var cache = {};
+
+  get(url) {
+    if (this.cache[url]) {
+      return Promise.resolve(this.cache[url]);
+    } else {
+      return fetch(url).then(function(response) {
+        return response.text()
```
Comment (pwalsh) [inline src2/api/index.js]: Why not `response.json()` so you also have the JSON object ready for later?

File: src2/api/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      }).then(function(text) {
+        return this.cache[url] = text
+      });
+    }
+  }
+
+  getJson(url) {
+    return this.get(url).then(JSON.parse)
```
Comment (pwalsh) [inline src2/api/index.js]: As above, if you use `response.json()` there you wouldn't need to parse here, or, is there a reason we need to do it like this?

File: src2/api/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  flush() {
+    this.cache = {}
+  }
+
+  buildUrl(endpoint, cube, path, params) {
+    params = params || {};
+    var api = endpoint;
+    api = endpoint[api.length - 1] == '/' ? api.slice(0, api.length - 1) : api;
+    api = api + '/cubes/' + cube + '/' + path;
```
Comment (pwalsh) [inline src2/api/index.js]: @borysyuk and cc @kravets-levko seeing as we are using ES6, try the new template strings. Example:

```
let api = `${api}/cubes/${cube}/path`
```
Reference: https://github.com/okfn/spend-publishing-dashboard/blob/master/app/utils.js#L19

**Comment by pwalsh** [comment]
@borysyuk code looks great. See my comments. Two other things for merge:

1. Tests (even just 1 or 2 for the basic API functionality will be enough, while you build the other components)
2. `src2` is not acceptable for merging, just call it `src` and do what ever you need to do locally to have the old and new accessible.

**Comment by pudo** [comment]
Can we make extra-sure that this doesn't break backward-compatibility and hence screw up spendb?

File: src2/api/index.js (hunk 2 above)
Comment (borysyuk) [inline src2/api/index.js]: @pwalsh i know about response.json()
`get` function can download any file for example csv
getJson - return response as json.

**Comment by pwalsh** [comment]
Ah @pudo you posted when I was writing to you here: https://github.com/openspending/babbage.ui/issues/16#issuecomment-190215530

File: src2/api/index.js (hunk 2 above)
Comment (pwalsh) [inline src2/api/index.js]: ok.

**Comment by pwalsh** [comment]
:+1:  @borysyuk ready to merge, but just use `lodash` and not `underscore`.

**Pull request merged**